src/main/java/org/fungover/thunder/Client.java

```diff
@@ -0,0 +1,46 @@
+package org.fungover.thunder;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Client {
+    private final String clientId;
+    private boolean connected;
+    private Set<String> subscribedTopics;
+
+    public Client(String clientId) {
+        this.clientId = clientId;
+        this.connected = false;
+        this.subscribedTopics = new HashSet<>();
+    }
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public boolean isConnected() {
+        return connected;
+    }
+
+    public void connect() {
+        if (!connected) {
+            // Establish a connection to the MQTT server
+            // Perform the MQTT connection handshake
+            System.out.println("Client " + clientId + " connected to server.");
+            connected = true;
+        } else {
+            System.out.println("Client " + clientId + " is already connected.");
+        }
+    }
```

Comment:
Client in our application means a class that we can create objects of that represents a connected client. So it isn't a client that connects to a mqtt broker/server. So when calling connect just mark this client as connected.
Logging of what happens might be better placed in the code that accepts new connections :)